libiocage/lib/Jail.py

```diff
@@ -409,11 +414,18 @@ def rename(self, new_name: str):
 
         current_id = self.config["id"]
         dataset = self.dataset
-        self.config["name"] = new_name  # validates new_name
+        current_dataset_name = dataset.name
+
+        self.config["id"] = new_name  # validates new_name
+        self.logger.debug(f"Renaming jail {current_id} to {new_name}")
         try:
-            dataset.rename(self.dataset_name)
+            new_dataset_name = self._dataset_name_from_jail_name
+            dataset.rename(new_dataset_name)
+            self.logger.verbose(
+                f"Dataset {current_dataset_name} renamed to {new_dataset_name}"
+            )
         except:
-            self.config["name"] = current_id
+            self.config["id"] = current_id
```

Comment:
what's up with the moving back and forth between `id` and `name`

Reply:
It's always ID. Name should always point to ID instead.

Reply:
aye